### ISSUE TITLE
[enhancement request] provide Postgres db support as an alternative to MongoDB

### ISSUE THREAD

**Issue by mariusa** [reactioncommerce/reaction#2411]
Please consider providing Postgres db support as an alternative to MongoDB. Thanks!

**Comment by jshimko** [comment]
Hey @mariusa.  Reaction is built on [Meteor](https://guide.meteor.com/) and Meteor is _deeply_ integrated with Mongo, so the only way that's going to be possible is to rewrite the entire API to use [GraphQL](http://graphql.org/).  Even once that process is done, someone would have to write the entire database interface layer using another database library.  You're probably looking at a couple months of hard work there.

However, we do happen to be working on the GraphQL part for the entire data API (which will exist in addition to the current DDP API).  But converting all of the data fetching to another database entirely is not on the roadmap at this point.  There's just not a strong enough case for that part being worth that much work.  

On the upside, we always like to keep things as extensible as possible, so we could easily make the entire GraphQL resolver layer pluggable so you could replace the Mongo fetching interface with something else.  As long as the returned data matches Reaction's core schema, GraphQL and the client side don't really care where that data came from (which is one of the main selling points for GraphQL).

So in short, GraphQL is definitely on the way, but Postgres won't be any time soon unless someone contributes it.  If you think that person could be you, absolutely get in touch with us in [the core Gitter channel](https://gitter.im/reactioncommerce/core) to discuss!

**Comment by mariusa** [comment]
Appreciate your feedback! Makes sense.

We'll look into it, although investing in customer facing features makes more sense given the effort. Feel free to close. At least it will show in results in future issue searches for Postgres
Thanks!
